pkg/controller/aws/network/internetgateway.go

```diff
@@ -0,0 +1,234 @@
+	if cr.Status.InternetGatewayID == "" {
+		return resource.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	req := e.client.DescribeInternetGatewaysRequest(&awsec2.DescribeInternetGatewaysInput{
+		InternetGatewayIds: []string{cr.Status.InternetGatewayID},
+	})
+	req.SetContext(ctx)
+
+	response, err := req.Send()
+
+	if err != nil {
+		if ec2.IsInternetGatewayNotFoundErr(err) {
+			return resource.ExternalObservation{
+				ResourceExists: false,
+			}, nil
+		}
+
+		return resource.ExternalObservation{}, errors.Wrapf(err, "Error describing InternetGateway with id: %v", cr.Status.InternetGatewayID)
+	}
+
+	// in a successful response, there should be one and only one object
+	if len(response.InternetGateways) != 1 {
+		return resource.ExternalObservation{}, errors.Errorf("Retrieved multiple InternetGateways for the given internetGatewaysId: %v", cr.Status.InternetGatewayID)
+	}
+
+	observed := response.InternetGateways[0]
+
+	// if non of the attachments are currently in progress, then the IG is available
+	isAvailable := true
+	for _, a := range observed.Attachments {
+		if a.State == awsec2.AttachmentStatusAttaching || a.State == awsec2.AttachmentStatusDetaching {
+			isAvailable = false
+			break
+		}
+	}
+
+	if isAvailable {
+		cr.SetConditions(runtimev1alpha1.Available())
+	}
+
+	cr.Status.InternetGatewayExternalState = cr.BuildExternalStateFromObservation(observed)
+
+	return resource.ExternalObservation{
+		ResourceExists:    true,
+		ConnectionDetails: resource.ConnectionDetails{},
+	}, nil
+}
+
+func (e *igExternal) Create(ctx context.Context, mgd resource.Managed) (resource.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.InternetGateway)
+	if !ok {
+		return resource.ExternalCreation{}, errors.New(errObjectNotInternetGateway)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	req := e.client.CreateInternetGatewayRequest(&awsec2.CreateInternetGatewayInput{})
+	req.SetContext(ctx)
+
+	ig, err := req.Send()
+	if err != nil {
+		return resource.ExternalCreation{}, errors.Wrap(err, "failed to create the InternetGateway resource")
+	}
+
+	cr.Status.InternetGatewayExternalState = cr.BuildExternalStateFromObservation(*ig.InternetGateway)
+
+	// after creating the IG, attach the VPC
+	aReq := e.client.AttachInternetGatewayRequest(&awsec2.AttachInternetGatewayInput{
+		InternetGatewayId: ig.InternetGateway.InternetGatewayId,
+		VpcId:             aws.String(cr.Spec.VPCId),
+	})
+	aReq.SetContext(ctx)
+
+	_, err = aReq.Send()
+
+	return resource.ExternalCreation{}, errors.Wrap(err, "failed to create the InternetGateway resource")
+}
+
+func (e *igExternal) Update(ctx context.Context, mgd resource.Managed) (resource.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha1.InternetGateway)
+	if !ok {
+		return resource.ExternalUpdate{}, errors.New(errObjectNotInternetGateway)
+	}
+
+	_ = cr
+
+	// TODO(soorena776): add more sophisticated Update logice, once we
+	// categorize immutable vs mutable fields (see #727)
```

Comment:
Perhaps this method should just return `resource.ExternalUpdate{}, nil` immediately if you don't think you'll implement update logic in this PR?